src/OrchardCore/OrchardCore.Mvc.Core/ModularApplicationModelProvider.cs

```diff
@@ -44,25 +44,22 @@ public void OnProvidersExecuted(ApplicationModelProviderContext context)
             {
                 var controllerType = controller.ControllerType.AsType();
 
-                var blueprint = _typeFeatureProvider.GetFeaturesForDependency(controllerType).FirstOrDefault();
+                var blueprint = _typeFeatureProvider.GetExtensionForDependency(controllerType);
 
-                if (blueprint != null)
```

Comment:
I don't know if this null check is necessary, but only remove it if you're sure it isn't.

Reply:
The `TypeFeatureProvider` will throw if no extension is available.

Reply:
Indeed.